buildtools/dep/dep.go

```diff
@@ -22,11 +43,32 @@ type Project struct {
 	Version  string
 }
 
-// A Lockfile contains the contents of a dep lockfile. Lockfiles are resolvers.
-type Lockfile struct {
-	Projects []Project
+//Resolve determines if the Gopkg.toml dep file has any rules that apply
+func (r Resolver) Resolve(importpath string) (pkg.Import, error) {
+	if r.isIgnored(importpath) {
+		return pkg.Import{}, buildtools.ErrNoRevisionForPackageOk
+	}
 
-	normalized map[string]pkg.Import // A normalized map of package import paths to revisions.
+	return r.Lockfile.Resolve(importpath)
+}
+
+//isIgnored finds packages to ignore while handling the possibility of a wildcard
+func (r Resolver) isIgnored(importpath string) bool {
```

Comment:
1. From an ontological perspective, I think this method makes more sense on the manifest.
2. I would be okay with this being public, especially for testing purposes.
3. [nit] Add a space between the `//` and the sentence.